hw/top_earlgrey/data/clocks.xdc

```diff
@@ -68,4 +72,6 @@ set_output_delay -clock clk_spi 5 [get_ports SPI_DEV_D1] -add_delay
 create_generated_clock -name clk_spi_in  -divide_by 1 -source [get_ports SPI_DEV_CLK] [get_pins top_*/u_spi_device/u_clk_spi_in_buf/gen_xilinx.u_impl_xilinx/gen_fpga_buf.gen_bufr.bufr_i/O]
 create_generated_clock -name clk_spi_out -divide_by 1 -source [get_ports SPI_DEV_CLK] [get_pins top_*/u_spi_device/u_clk_spi_out_buf/gen_xilinx.u_impl_xilinx/gen_fpga_buf.gen_bufr.bufr_i/O] -invert
 
-set_clock_groups -group ${clks_10_unbuf} -group ${clks_48_unbuf} -group ${clks_aon_unbuf} -group clk_io_div2 -group clk_io_div4 -group lc_jtag_tck -group rv_jtag_tck -group {clk_spi clk_spi_in clk_spi_out} -group sys_clk_pin -asynchronous
+
+
+set_clock_groups -group ${clks_10_unbuf} -group ${clks_48_unbuf} -group ${clks_aon_unbuf} -group clk_io_div2 -group clk_io_div4 -group usbdev_clk -group lc_jtag_tck -group rv_jtag_tck -group {clk_spi clk_spi_in clk_spi_out} -group sys_clk_pin -asynchronous
```

Comment:
`usbdev_clk` and `clks_48_unbuf` should be placed in the same group, shouldn't they?

Maybe do something like below and use that group instead of the other two?
```
set usbdev_clk_group [get_clocks -include_generated_clocks ${clks_48_unbuf}]
```

Reply:
i'm actually going to take these out and experiment a bit locally to see what the problematic path is.

so to your question, they actually "shouldn't" in this case, the fact they were was causing the hold time violation, since there was a synchronous path to a pulse sync module between what was effectively and early clock tree and a late clock tree. 

the `clks_48_unbuf` should be handling some of the calibration logic inside `ast`, and that one currently has an `async` assumption (based on design) with the `usbdev` clock interface.  Based on what I know of the design though, the direction of communication shouldn't have a hold time issue (since it's going from late tree to early tree, not the other way around), so I need to investigate a bit.

I did this fully async thing mostly to prove to myself this was the issue before I had a chance to look closely.

Reply:
Ahh, okay, then yeah, at least swapping out these super blunt set_clock_group constraints for something targeted sounds better. :joy:

Thanks for explaining! If there is communication between them, the extra BUFG is going to cause considerable skew, which might be part of what's causing headaches. :thinking: